eva_submission/eload_status.py

```diff
@@ -0,0 +1,358 @@
+            "from project_analysis pa "
+            "join analysis a on pa.analysis_accession=a.analysis_accession "
+            "left join assembly_set at on at.assembly_set_id=a.assembly_set_id "
+            "left join analysis_file af on af.analysis_accession=a.analysis_accession "
+            "join file f on f.file_id=af.file_id "
+            f"where f.file_type='VCF' and pa.project_accession='{self.project}' "
+        )
+        if analysis:
+            query += f"and pa.analysis_accession='{analysis}'"
+        else:
+            query += "order by pa.project_accession, pa.analysis_accession"
+        filenames = []
+        current_analysis = current_assembly = current_tax_id = None
+        with self.metadata_connection_handle as conn:
+            for project, analysis, assembly, tax_id, filename in get_all_results_for_query(conn, query):
+                if analysis != current_analysis:
+                    if current_analysis:
+                        yield current_analysis, current_assembly, current_tax_id, filenames
+                    current_analysis = analysis
+                    current_assembly = assembly
+                    current_tax_id = tax_id
+                    filenames = []
+                filenames.append(filename)
+            yield current_analysis, current_assembly, current_tax_id, filenames
+
+    def get_taxonomy_for_project(self):
+        taxonomies = []
+        query = f"select distinct taxonomy_id from evapro.project_taxonomy where project_accession='{self.project}'"
+        with self.metadata_connection_handle as conn:
+            for tax_id, in get_all_results_for_query(conn, query):
+                taxonomies.append(tax_id)
+        if len(taxonomies) == 1:
+            return taxonomies[0]
+        else:
+            self.error(f'Cannot retrieve a single taxonomy for project {self.project}. Found {len(taxonomies)}.')
+
+    def check_accessioning_was_done(self, analysis, filenames):
+        """
+        Check that an accessioning file can be found in either noah or codon (assume access to both filesystem)
+        It parses and provide a 1000 submitted variant accessions from that project.
+        """
+        accessioning_reports = self.get_accession_reports_for_study()
+        accessioned_filenames = [self.get_accession_file(f) for f in filenames]
+        if not accessioning_reports:
+            return []
+        if len(accessioning_reports) == 1:
+            accessioning_report = accessioning_reports[0]
+        elif len([r for r in accessioning_reports if r in accessioned_filenames]) == 1:
+            accessioning_report = [r for r in accessioning_reports if os.path.basename(r) in accessioned_filenames][
+                0]
+        elif len([r for r in accessioning_reports if analysis in r]) == 1:
+            accessioning_report = [r for r in accessioning_reports if analysis in r][0]
+        elif accessioning_reports:
```

Comment:
I don't think you need this check since it's done in line 248 above (unless I missed something)

Reply:
Right I've actually remove the line above as this one makes more sense to me